sdk/anomalydetector/ai-anomaly-detector/src/AnomalyDetectorClient.ts

```diff
@@ -29,6 +29,10 @@ import { logger } from "./logger";
 import { createSpan } from "./tracing";
 import { CanonicalCode } from "@opentelemetry/api";
 
+export type DetectEntireSeriesOptions = OperationOptions;
+export type DetectLastPointOptions = OperationOptions;
+export type DetectChangePointOptions = OperationOptions;
```

Comment:
I wonder if we should add `OperationOptions` as a valid name for the options bag. It feels excessive to create new types which are only aliases for OpertaionOptions and do not really extend it

Reply:
Hmm, I am not sure. One upside of creating new types is that extending them later on feels natural without having to worry about creating new types then. Either way, I think we should be consistent across our packages. The [guidelines](https://azure.github.io/azure-sdk/typescript_design.html#ts-apisurface-serviceclient) does not specifically mention how to name the options parameter but in their examples all options params have a prefix with the method name.

Reply:
I agree that it feels excessive and *most* of the time, the type gets inferred by assignment, so we don't have to worry about folks having trouble upgrading, but I'm not sure if it hurts anything to have the specific type name reserved in case we add options later.

I could be swayed either way on this, but so far we've gone with always providing a type alias to future-proof.

Reply:
I don’t have a strong preference. However I do agree on consistency so I would suggest that if we keep recomending the aliases we should mention it in the guidelines. Otherwise we should add OperationOptions as a valid name.

This aliasing wouldn’t be considered a breaking change, would it?

Reply:
@joheredi with structural typing, type names do not matter when checking two types for equality so I do not think this type rename is a breaking change. @willmtemple, @bterlson, @xirzec can you think of a scenario where this could be a breaking change?

Reply:
The above is not breaking AFAIK. The reason we like doing the above aliasing is because otherwise maybe someone writes:

```ts
const myOptions: OperationOptions = { /* options here */ };
```

but then in a later version we add some actual options to that specific operation and then they have to import the new name and update their code, whereas if they are already pulling in a friendly type alias, even if we change that to a proper interface later, they can just use IntelliSense to see the new properties.

It's pretty niche though.

Reply:
I opened this PR to add it as a guideline: https://github.com/Azure/azure-sdk/pull/1885.